구현 과제 - 로그인/회원가입 모달 - 의진

timetam24 opened the pull request:
로그인/회원가입 첫 번째 레이아웃을 구현하였습니다.

timetam24 committed:
[#11] 로그인 모달 및 기능 구현

AYFG commented:
와~로그인 성공 alert 상상도 못했어요😆 헤더에 word-spacing으로 피그마랑 완전 똑같이 구현하려하신 꼼꼼함도 보이네요
h2태그에 아스키코드로 만드셨네요 저는 그냥  " ) " 로 했는데 빨갛게 변하더라고요 덕분에 배웠습니다..
의진님 코드보고 많이 배웠습니다 감사해요!!!👍👍👍👍

timetam24 commented:
> 와~로그인 성공 alert 상상도 못했어요😆 헤더에 word-spacing으로 피그마랑 완전 똑같이 구현하려하신 꼼꼼함도 보이네요 h2태그에 아스키코드로 만드셨네요 저는 그냥 " ) " 로 했는데 빨갛게 변하더라고요 덕분에 배웠습니다.. 의진님 코드보고 많이 배웠습니다 감사해요!!!👍👍👍👍

신경 쓴 부분 꼼꼼하게 캐치해주셨네요 감사합니다 재웅님!! 저도 앞으로 많이 보고 배우겠습니다😊 리뷰 다시 한번 감사드려요!!👍

timetam24 committed:
[#11] 로그인 모달닫기 키보드 포커싱 문제해결

aicul313 commented:
우와 자스까지 넣으시다니 대단해요 😚👍 h2 태그 아스키코드도 그렇고 헤더 h1 태그에 word-spacing으로 글자 간격은 생각도 못했는데 꼼꼼함이 보이는 것 같아요 대단해요 !!! 의진 님 word-spacing 넣으신 것보고 제 코드에도 적용해보아야겠어요 !! 고생하셨습니다 의진 님 🥰

timetam24 commented:
> 우와 자스까지 넣으시다니 대단해요 😚👍 h2 태그 아스키코드도 그렇고 헤더 h1 태그에 word-spacing으로 글자 간격은 생각도 못했는데 꼼꼼함이 보이는 것 같아요 대단해요 !!! 의진 님 word-spacing 넣으신 것보고 제 코드에도 적용해보아야겠어요 !! 고생하셨습니다 의진 님 🥰

와아 세세한 리뷰 감사합니다 현지님😊 저도 앞으로 현지님 코드 보고 많이 배울게요!! 같이 좋은 방향으로 성장했으면 좋겠습니다❤️

YennieJ commented on login-modal/style.css:
```diff
@@ -0,0 +1,311 @@
+  height: 100vh;
+  font-family: "SpoqaHanSansNeo-Medium";
+  font-size: 16px;
+  color: #767676;
```
body에 color자체를 가장 많이 사용한 색으로 하셨네요!!👍

YennieJ commented on login-modal/index.html:
```diff
@@ -0,0 +1,72 @@
+              class="user-pwd"
+              placeholder="비밀번호"
+              autocomplete="off"
+            />
```
웹 접근성을 위한 label 활용을 고려해보면 좋겠어요

YennieJ commented on login-modal/style.css:
```diff
@@ -0,0 +1,311 @@
+  width: 100%;
+  border-radius: 5px;
+  border: 1px solid #c4c4c4;
+  padding: 13px 0 13px 16px;
```
실제 서비스라면 html input 태그에 아이디와 비밀번호에   maxlength를 사용할 가능성이 높지만 현재는 그렇지 않으니
오른쪽에도 패딩이 있으면 좋을꺼같습니다:)

YennieJ commented on login-modal/style.css:
```diff
@@ -0,0 +1,311 @@
+  text-align: center;
+}
+
+.login-service .join-member::after {
+  content: "";
+  display: inline-block;
+  width: 1px;
+  height: 14px;
+  margin-left: 13px;
+  margin-right: 13px;
+  vertical-align: -2px;
+  background-color: #767676;
+}
```
``` <a href="#" class="join-member">``` 자체에 after를 주게되면 after도 a태그 영역으로 묶이게 되기 때문에 "|" 가상요소 또한 a태그가 됩니다.
이런 경우는 ul>li>a로 마크업을 구성하고 li에 가상요소를 주면 해당 문제는 해결된다고 저희 멘토님께 답변을 받았습니다 :) 
참고하시면 좋을꺼같아요

timetam24 replied:
제가 마지막까지 고민하던 문제였는데 li에 가상요소를 주면 되었군요! 깔끔하게 회원가입에만 키보드 포커싱 되니까 훨씬 보기 좋네요 감사합니다!!

YennieJ commented on login-modal/style.css:
```diff
@@ -0,0 +1,311 @@
+  display: block;
+  width: 28px;
+  height: 28px;
+  background-image: url(./images/icon-sprites.png);
```
스프라이트를 사용하니 코드가 훨씬 깔끔하네요👍

YennieJ commented on login-modal/app.js:
```diff
@@ -0,0 +1,64 @@
+  };
+
+  if (!userId.value.trim()) {
+    // 아이디 미입력 시
+    userId.classList.add("focus");
+    createErrorMsg(
+      firstErrorMsg,
+      "first-error-msg",
+      "아이디를 입력해주세요.",
+      userPwd
+    );
+    return;
+  } else if (userId.value.trim()) {
+    // 아이디 입력 시
+    userId.classList.remove("focus");
+    removeErrorMsg(firstErrorMsg);
+
+    if (!userPwd.value.trim()) {
+      // 아이디 입력 & 비밀번호 미입력 시
+      userPwd.classList.add("focus");
+      createErrorMsg(
+        secondErrorMsg,
+        "second-error-msg",
+        "비밀번호를 입력해주세요.",
+        userPwd.nextSibling
+      );
+      return;
+    } else if (userPwd.value.trim()) {
+      // 아이디 입력 & 비밀번호 입력 시
+      userPwd.classList.remove("focus");
+      if (userId.value !== "weniv07" && userPwd.value !== "frontend07!") {
+        secondErrorMsg.innerText = "아이디 혹은 비밀번호가 일치하지 않습니다.";
+        return;
+      } else {
+        removeErrorMsg(secondErrorMsg);
+        alert("로그인 성공!");
+      }
+    }
+  }
+};
```
if문.. 설정하기 너무 힘들죠 :(
중첩되는 if문이 많아서 제 나름대로 한번 줄여봤어요!
참고만 해주세요 :)
주석으로 달아 놓은 코멘트는 수정이 필요할꺼 같습니다!
```
  if (!userId.value.trim()) {
    userId.classList.add("focus");
    createErrorMsg(
      firstErrorMsg,
      "first-error-msg",
      "아이디를 입력해주세요.",
      userPwd
    );
    return;
  } else if (userId.value.trim() && !userPwd.value.trim()) {
    userId.classList.remove("focus");
    removeErrorMsg(firstErrorMsg);
    userPwd.classList.add("focus");
    createErrorMsg(
      secondErrorMsg,
      "second-error-msg",
      "비밀번호를 입력해주세요.",
      userPwd.nextSibling
    );
    return;
  } else if (userId.value !== "weniv07" && userPwd.value !== "frontend07!") {
    userId.classList.remove("focus");
    userPwd.classList.remove("focus");
    // createErrorMsg를 사용한 이유는 처음에 아이디와 비밀번호가 일치하지 않을 때 경고문이 나타나지 않음.
    createErrorMsg(
      secondErrorMsg,
      "second-error-msg",
      "아이디 혹은 비밀번호가 일치하지 않습니다.",
      userPwd.nextSibling
    );
  } else {
    removeErrorMsg(secondErrorMsg);
    alert("로그인 성공!");
  }
```

timetam24 replied:
비밀번호 미입력과 아이디,비밀번호 불일치 부분에서 if문 작성하느라 머리 부여잡았는데 더 간결한 방법이 있었네요! 코드 보면서 많이 배웠어요 감사합니다 예진님!!🥰

YennieJ reviewed (commented):
의진님 스프라이트부터 순수 자바스크립트, 시맨틱 태그, 주석까지
깔끔한 코드 잘 봤습니다! 공부가 많이 됐어요 :)
고생하셨습니다!

YennieJ commented on login-modal/index.html:
```diff
@@ -0,0 +1,72 @@
+            />
+            <input type="checkbox" id="inpHold" class="inp-hold txt-hide" />
+            <label for="inpHold" class="labl-hold">로그인 상태 유지</label>
+            <button class="login-btn">로그인</button>
```
type 이 추가 됐으면 좋겠어요:)

kyokyoYa reviewed (commented):
JS에서 if 문을 어떻게 사용하는 궁금했는데 의진 님 코드를 보면서 해소됐네요!
전반적으로 잘하신 코드라고 생각합니다.
특히 class명을 직관적으로 잘 주셔서 많이 배웠습니다🙌

timetam24 committed:
[#11] 웹 접근성 위한 label 추가 & 가상요소 적용 태그 변경

timetam24 committed:
[#11] 중첩 if문 수정

timetam24 committed:
[#11] Fix: "아이디와 비밀번호 불일치 안내 메세지 오류 수정"

1. 로그인 성공 후 불일치 안내 메세지가 삭제되지 않는 오류 수정
2. 아이디 미입력 시 기존의 불일치 안내 메세지가 삭제되지 않고 아이디 입력 안내 메세지와 동시 출력되는 오류 수정

timetam24 committed:
[#11] Fix: "아이디 미입력 안내 메세지 오류 수정"

아이디 또는 비밀번호 불일치 시 기존의 아이디 미입력 안내 메세지와 focus가 사라지지 않고 불일치 안내 메세지와 동시 출력되는 오류 수정

timetam24 committed:
[#11] 로그인 성공 시 비밀번호 focus 삭제

timetam24 committed:
[#11] Fix: "비밀번호 focus 오류 수정"

아이디 미입력 시 기존의 비밀번호 focus가 삭제되지 않고 아이디 미입력 안내 메세지 그리고 아이디 focus와 함께 출력되는 오류 수정

timetam24 committed:
[#11] Refactor: "중복 코드 함수로 추출"

YennieJ commented on login-modal/app.js:
```diff
@@ -22,10 +22,15 @@ const loginBtnHandler = (event) => {
     }
   };
 
+  const checkFocusAndErrorMsg = (input, errorMsg) => {
+    input.classList.remove("focus");
+    removeErrorMsg(errorMsg);
+  };
+
```
키야 내부함수를 바로 만드시다니 증말 최고애요👍

timetam24 replied:
ㅎㅎㅎ 감사합니다❤️
